### PR TITLE
Batch prefetch in large copy operations

### PR DESCRIPTION
**Pull request by hoytak**
This PR:
- Reworks the _copy logic to separate out the cases, making debugging and modifying and optimizing much easier.
- Adds a deduplication stage that fetches batch dedup shards for any existing destination files with source data larger than 10MB, as well as the 16 largest files in the batch.

### CODE REVIEW

File: python/pyxet/pyxet/cli.py [hunk 1]
```diff
@@ -149,114 +184,123 @@ def _copy(source, destination, recursive = True, _src_fs=None, _dest_fs=None):
...
+        if not recursive:
+            raise ValueError(
+                "Specify recursive flag '-r' to copy directories.")
+        try:
+            dest_info = dest_fs.info(dest_path)
+            if dest_info['type'] == 'directory':
+                # The destination directory is actually the source 
+                _, end_component = os.path.split(src_path)
+                processed_path = os.path.join(dest_path, end_component)
+            else:
```
[the remark this answers is not in the document]
Comment (hoytak) [reply python/pyxet/pyxet/cli.py:215]: Good catch, will update that.

[the remark this answers is not in the document]
Comment (hoytak) [reply python/pyxet/pyxet/cli.py:215]: So looking back through the logic here, _isdir is true for branches too if copying to/from a xet endpoint.

[the remark this answers is not in the document]
Comment (seanses) [reply python/pyxet/pyxet/cli.py:215]: But I think dest_info is obtained from calling [info()](https://github.com/xetdata/pyxet/blob/3195ed0d08a75c1368fcfdbe5dba8d91cdda83b0/python/pyxet/pyxet/file_system.py#L160) and info['type'] [can be "directory" or "branch"](https://github.com/xetdata/pyxet/blob/3195ed0d08a75c1368fcfdbe5dba8d91cdda83b0/python/pyxet/pyxet/file_system.py#L126), so probably we can directly use [isdir](https://github.com/xetdata/pyxet/blob/3195ed0d08a75c1368fcfdbe5dba8d91cdda83b0/python/pyxet/pyxet/file_system.py#L118) from file_system.py

**Comment by hoytak** [comment]
Supplanted by https://github.com/xetdata/pyxet/pull/97

**Pull request closed without merging**